elements/lisk-transactions/README.md

```diff
@@ -2,6 +2,81 @@
+5. Validated transactions go to the `prepare` step defined in the transaction class, which to limit the I/O database operations prepares all the information relevant to properly `apply` or `undo` the transaction. The store with the prepared data is a parameter of the mentioned methods.
+6. Delegates are forging the valid transactions into the blocks and broadcasting the blocks to the network. Each network node performs the `apply` and `applyAsset` steps after the successful `validate` step.
+7. It is probable, especially shortly after a block is applied, that due to the decentralized network conditions a node does the `undo` step and the block containing all of the included transactions get reverted in favour of a competing block.
+
+While implementing a custom transaction, it is necessary to implement some of the mentioned steps. For most of them, a base transaction implements a default behaviour. As you feel more confident in using Lisk SDK, it is more likely for developers to override most of the base transaction methods, so the implementation is well-tailored and implemented with the best possible performance to the application's use case.
+
+### Interface
+
+#### Required
+
+All of the abstract methods and properties on the base transaction's interface are required to implement. Those are:
+
+##### TYPE
+
+> static TYPE: number
+
+The hallmark of a transaction. Override this static parameter with any number, keeping in mind that the first 10 types (0-9) are reserved for the default transactions.
+
+##### applyAsset
+
+> applyAsset(store: StateStore): ReadonlyArray<TransactionError>
+
+The business use-case of a transaction is implemented in `applyAsset` method. Apply all of the necessary changes from the received transaction to the affected account(s) by calling `store.set`. Call `store.get` to get all of the relevant data. The transaction that you're currently processing is the function's context (like `this.amount`).
+Invalidate the transaction by pushing an error into the result array.
+
+##### undoAsset
+
+> undoAsset(store: StateStore): ReadonlyArray<TransactionError>
+
+The invert of `applyAsset`. Roll-back all of the changes to the accounts done in the `applyAsset` step.
+
+##### validateAsset
+
+> validateAsset(): ReadonlyArray<TransactionError>
+
+Before a transaction reaches the apply step it gets validated. Check the transaction's asset correctness from the schema perspective (no access to StateStore here).
+Invalidate the transaction by pushing an error into the result array.
+
+##### prepare
+
+> prepare(store: StateStorePrepare): Promise<void>
+
+Prepare the relevant information about the accounts, which will be accessible in the later steps during the `apply` and `undo` steps.
+
+#### Performance
+
+To increase your application's performance, you should override the following functions: `verifyAgainstTransactions`, `assetFromSync`, `fromSync`.
```

Comment:
`assetFromSync`, `fromSync` these are not for performance. (it's more for legacy sync format)

Reply:
Changed the title to "additional methods"